StatBot/Database/PostgresRepositories/PostgresStatsRepository.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace StatBot.Database.PostgresRepositories
+{
+    public class PostgresStatsRepository : IRepository<Statistic>
+    {
+        private StatDbContext _context;
+
+        public PostgresStatsRepository()
+        {
+            _context = new StatDbContext();
+        }
+
+        public void Add(Statistic stat)
+        {
+            var count = GetAll().ToList().Count;
+            stat.Id = count + 1;
+
+            _context.Statistics.Add(stat);
+        }
+
+        public void Delete(int id)
+        {
+            var stat = _context.Statistics.Find(id);
+            if (stat != null)
+            {
+                _context.Statistics.Remove(stat);
+            }
+        }
+
+        public void Dispose()
+        {
+            _context.Dispose();
+            GC.SuppressFinalize(this);
+        }
+
+        public IEnumerable<Statistic> GetAll()
+        {
+            return _context.Statistics;
+        }
+
+        public Statistic GetById(int id)
+        {
+            return _context.Statistics.Find(id);
+        }
+
+        public void Save()
+        {
+            _context.SaveChanges();
+        }
+
+        public void Update(Statistic stat)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename parameter 'stat' to 'o' to match the interface declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://172.21.13.172:9091/coding_rules#rule_key=csharpsquid%3AS927)